mopidy/backend.py

```diff
@@ -122,23 +166,28 @@ def get_images(self, uris):
         """
         return {}
 
-    def lookup(self, uri):
+    def lookup(self, uri: Uri) -> Dict[Uri, List[Track]]:
         """
         See :meth:`mopidy.core.LibraryController.lookup`.
 
         *MUST be implemented by subclass.*
         """
         raise NotImplementedError
 
-    def refresh(self, uri=None):
+    def refresh(self, uri: Optional[Uri] = None) -> None:
         """
         See :meth:`mopidy.core.LibraryController.refresh`.
 
         *MAY be implemented by subclass.*
         """
         pass
 
-    def search(self, query=None, uris=None, exact=False):
+    def search(
+        self,
+        query: Query[SearchField] = None,
```

Comment:
Why don't these params get wrapped in `Optional[]` like in `get_distinct()` ?

Reply:
From what I can see in the Core API, `query` is always set, and Core itself crashes if it isn't a dict, so I'll remove `= None` instead of adding `Optional`.

However, I'm adding the missing `Optional` on the `uris` argument.